Add a total-lines segment

hinrik opened the pull request:
This is equivalent to `%L` in Vim's status line.

Uses https://melpa.org/#/total-lines

deb0ch commented:
Hello, thanks for your contribution !

Great to see such a feature, I've been missing something like this for the longest time without ever taking the time to dig into it ! I wonder why nobody has implemented it before, in 45 years of Emacs.

As your package seems fairly new I will take some time for testing it locally 😸

As of the form, having a new segment available for this might indeed be relevant, but I'm thinking that it might be more beneficial to have it replace the `buffer-size` segment (size of the current buffer in kilobytes) when available, to avoid cluttering the Spaceline with an extra segment.

I wonder if the size in kilobytes is more meaningful to anyone than the number of lines in the buffer when it comes to editing text files.

At least personally I would see your segment best as a replacement for `buffer-size` 😸

hinrik commented:
Yeah, I've also wondered why this isn't something that Emacs itself keeps track of, like Vim does. Fortunately it was possible to implement it after the fact in a performant way.

The `buffer-size` certainly seems interesting when editing text than the total number of lines. Would be nice to see the line total used instead!

hinrik commented:
I fixed some bugs with the mode, let me know if you uncover any issues.

deb0ch commented:
Hey, just found a bug in your mode making it not work for me. Opened a ticket in your repo: https://github.com/hinrik/total-lines/issues/1

hinrik commented:
Just pushed a fix for this.

hinrik commented:
Any interest in this?

hinrik commented:
@deb0ch I've replaced `buffer-size` with `total-lines` as you suggested. I also replaced `buffer-position` with a new segment based on line count instead of the buffer size, to match.

These changes are straightforward, but I wonder whether it would make sense to go further. Now that the total # of lines are in the modeline, wouldn't it make more sense to group the total and current line numbers together? For instance, my modeline contains the following segment:

```
(spaceline-define-segment line-total-column
  "Line/Total,Column"
  (format
   "%s/%d,%s"
   (format-mode-line "%l")
   total-lines
   (format "%-2d" (1+ (current-column)))))
```

deb0ch commented:
Closed in favor of #229
